wrapper.go

```diff
@@ -345,6 +345,11 @@ func (r *resticWrapper) getCommandArgumentsFilter(command string) argumentsFilte
 	return nil
 }
 
+func (r *resticWrapper) containsArguments(arguments []string, subset ...string) (found bool) {
+	found = len(r.validArgumentsFilter(subset)(arguments, true)) > 0
```

Comment:
I can't figure out why yet, but it seems `containArgument` fails to recognize `--retry-lock` as a valid argument for the `backup` command. Funny enough it works for the `init` command:

```
2023/08/08 22:23:23 resticprofile 0.23.0-dev compiled with go1.20.7
2023/08/08 22:23:23 loading: examples/dev.yaml
2023/08/08 22:23:23 files in configuration are relative to "examples"
2023/08/08 22:23:23 preventing the system from sleeping
2023/08/08 22:23:23 memory available: 13722MB
2023/08/08 22:23:23 setting process priority to 10
2023/08/08 22:23:23 restic 0.16.0
2023/08/08 22:23:23 setting up environment variable: TMP=/tmp
2023/08/08 22:23:23 profile 'self': initializing repository (if not existing)
2023/08/08 22:23:23 unfiltered extra flags:
2023/08/08 22:23:23 unfiltered command: init --cleanup-cache --password-file=examples/key --repo=/Volumes/RAMDisk/self --retry-lock=3m --verbose=1
2023/08/08 22:23:23 setting up environment variable: TMP=/tmp
2023/08/08 22:23:23 starting command: restic init --cleanup-cache --password-file=examples/key --repo=/Volumes/RAMDisk/self --retry-lock=3m --verbose=1
2023/08/08 22:23:23 setting up environment variable: TMP=/tmp
2023/08/08 22:23:23 profile 'self': starting 'backup'
2023/08/08 22:23:23 unfiltered extra flags:
2023/08/08 22:23:23 unfiltered command: backup --cleanup-cache --exclude="/**/.git/" --password-file=examples/key --repo=/Volumes/RAMDisk/self --verbose=1 /go/src/github.com/creativeprojects/resticprofile
2023/08/08 22:23:23 setting up environment variable: TMP=/tmp
2023/08/08 22:23:23 starting command: restic backup --cleanup-cache --exclude="/**/.git/" --password-file=examples/key --repo=/Volumes/RAMDisk/self --verbose=1 /go/src/github.com/creativeprojects/resticprofile
```

(VSCode bug, it did put my comment on the wrong line to start with)

Reply:
This means you might have found something. In my setup it works for backup as well. I'll check it with "dev.yaml"

Reply:
might have found the problem

Reply:
Good catch, should be fixed now. To be honest this filter should be moved before or into the shell args composition (which also should be updated to better partition arguments positions). The current solution is more a hack and I fell into the pitfall that it passes values (which may or may not belong to flags).